subsys/bluetooth/host/hci_core.c

```diff
@@ -5034,79 +5057,100 @@ static bool valid_adv_param(const struct bt_le_adv_param *param)
 		}
 	}
 
-	if (param->interval_min > param->interval_max ||
-	    param->interval_min < 0x0020 || param->interval_max > 0x4000) {
-		return false;
+	if ((param->options & BT_LE_ADV_OPT_DIR_MODE_LOW_DUTY) ||
+	    !dir_adv) {
+		if (param->interval_min > param->interval_max ||
+		    param->interval_min < 0x0020 ||
+		    param->interval_max > 0x4000) {
+			return false;
+		}
 	}
 
 	return true;
 }
 
-int bt_le_adv_start(const struct bt_le_adv_param *param,
-		    const struct bt_data *ad, size_t ad_len,
-		    const struct bt_data *sd, size_t sd_len)
+static inline bool name_is_present_in_adv_data(const struct bt_data *ad,
+					       size_t ad_len)
+{
+	int i;
+
+	for (i = 0; i < ad_len; i++) {
+		if (ad[i].type
+			== BT_DATA_NAME_COMPLETE ||
+		    ad[i].type
+			== BT_DATA_NAME_SHORTENED) {
+			return true;
+		}
```

Comment:
```
                for (i = 0; i < ad_len; i++) {
                        if (ad[i].type == BT_DATA_NAME_COMPLETE ||
                            ad[i].type == BT_DATA_NAME_SHORTENED) {
                                return true;
                        }
                }
```

Reply:
Yes, sorry. I forgot about the main reason why I created this helper function